ITGlueAPI/Internal/APICalls.ps1

```diff
@@ -37,19 +43,48 @@ function Invoke-ITGlueRequest {
     try {
         $ITGlue_Headers.Add('x-api-key', (New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList 'N/A', $ITGlue_API_Key).GetNetworkCredential().Password)
 
-        $parameters = @{
-            'Method' = $Method
-            'Uri' = $ITGlue_Base_URI + $ResourceURI + $query_string
-            'Headers' = $ITGlue_Headers
-            'Body' = $body
+        $page = 0
+        
+        do {
+            if($AllResults) {
+                $page++
+                if(-not $QueryParams) { $QueryParams = @{} }
+                $QueryParams['page[number]'] = $page
+            }
+
+            $query_string = ConvertTo-QueryString -QueryParams $QueryParams
+
+            $parameters = @{
+                'Method' = $Method
+                'Uri' = $ITGlue_Base_URI + $ResourceURI + $query_string
+                'Headers' = $ITGlue_Headers
+                'Body' = $body
+            }
+
+            $api_response = Invoke-RestMethod @parameters -ErrorAction Stop
+
+            if($AllResults) {
+                $result.data += $api_response.data
+            } else {
+                $result = $api_response
+            }
+
+        } while($AllResults -and $api_response.meta.'total-pages' -and $page -le ($api_response.meta.'total-pages'))
```

Comment:
You're making me second guess myself here, but I believe I have to increment before the call or we would still end up with an off-by-one due to the while condition failing prematurely. And due to not knowing the total pages before the first call to the API I cannot do a normal while loop which would be more readable. You are correct about there being an off-by-one though, it should be `-lt`. I have fixed it.

Reply:
Ah, I just realized that I wasn't entirely correct either. If $page is initialized to 1 and incremented at the end of the loop, the condition would need to be `-le` instead of `-lt`.

I think it's fine the way it is; it's plenty readable. Thanks for your efforts!